### ISSUE TITLE
Shared folder: Guest not seeing files if it's attempted to access them before they're created

### ISSUE THREAD

**Issue by TomBZombie** [mitchellh/vagrant#6432]
I realise this may not be a vagrant bug itself, but a configurtion issue but hopefully someone here can help

Using the following setup: Apache is running in the guest with its server root set to /srv/http, this is a synched folder which points to ./public_html on the host.

While most of the time it works as expected, the following steps causes an issue

1) Navigate to a file that doesn't exist localhost:8080/test2.css -- shows a 404 error as expected but correctly connects to the guest which is serving the error

2) Create test2.css with some content and place it in public_html

3) Reload localhost:8080/test2.css -- Still shows a 404 error even though the file now exists

4) To debug, run vagrant ssh and then ls /srv/http. Which shows:
```
    ls: cannot access test2.css: No such file or directory
```

So it's seeing the file, sort of but it shows without any permissions:
```
    -????????? ? ?          ?            ? test2.css
    -rw-rw-r-- 1 vagrant    7 Oct 23 11:13 test3.css
```
If I then re-save the file as test3.css, a file that hasn't yet been accessed it works perfectly. E.g. on the host, save the file I had open as test3.css and then navigate to it, it works as expected!

Any ideas? On why this might be?

In short: If apache has tried to read a file that doesn't exist, creating that file will then cause it to have invalid permissions. If apache has never tried to read the file before, it can be created and work as expected.

Thanks for any help, I'm really confused by this!

edit: I have simplified the steps to reproduce:

1) In guest navigate to /srv/http

2) Type stat newfile.css -- correctly displays that the file does not exist

3) Create newfile.css on host

4) Type ls on guest and the file has ???? permissions and gives the "No such file or directory" error. Is it some kind of stat cache issue?

**Comment by sethvargo** [comment]
Hi @TomBZombie 

Thank you for opening an issue. This is definitely quite strange. I just tried on Ubuntu 14.04 and I was unable to reproduce this issue, but I could be doing something different or have a different setup.

With your edit, I do not believe this is an issue with Vagrant, since you are able to reproduce this issue simply by running commands on the guest. Most likely, this is either a bug in virtualbox or a bug in the shared folder implementation you are using. You might want to give the rsync shared folders a try and see if that solves your issue. It would also be helpful to know if the behavior exists when you perform these operations on a non-shared volume.

**Comment by TomBZombie** [comment]
I've actually found the source of the issue after much trial and error I found it's a kernel bug that affects at least 4.1 - 4.2.3. Upgrading to Kernel 4.2.4 solves the issue. This does affect Ubuntu 15.10 because it uses an affected kernel.
